### ISSUE TITLE
Error: list: Packet length 49989 exceeds max length of 34000

### ISSUE THREAD

**Issue by ardean** [theophilusx/ssh2-sftp-client#343]
Maybe I was wrong in my first issue at the ssh2 library.

They do not answer me there, so I ask here if you have an idea what the best solution would be.
My Issue is that I can not list a large Folder. I will reach the packet max length, which is hardcoded.

Here is the original issue https://github.com/mscdex/ssh2/issues/1048

What do you think is the best solution?

Thank you for your help.

**Comment by theophilusx** [comment]
As this is a limit imposed by ssh2, I think you need to persist with
trying to get it resolved in that package. There is nothing I can think
of which would effectively address the issue.

Note that I'm not 100% convinced that your analysis is correct. I"m
pretty sure that the data is broken into multiple packets and not trying
to put it all in one packet. What I suspect is actually going on is that
the client and server are not negotiating consistent packet sizes. The
server is sending a larger packet than the client is expecting. Note
that I have used the client to list directories with many more files
than your example with no problems.

First thing I would do is use the openSSH CLI program to query the
remote server and get a directory listing using the -v switch to
increase logging (you may need multiple 'v's to increase the logging
enough to see relevant data). I would then write a very simple ssh2
script to list the remote directory and add a logging function to see
what it is doing and compare to the logs from openssh's sftp cli. Once
you have an example script which is able to reproduce the problem only
using ssh2, add that to the issue you have in the ssh2 issue log.

You don't mention node version, platform or anything about the remote
sftp server in your issue. I would make sure to add this information as
it could be relevant/important. For example, there are known sftp
servers which have problems/bugs. 

I would also try connecting from a completely different system - ideally
one which involves different network/firewall etc. I would also try
connecting to a different server which has a large number of files to
see if you can reproduce the issue. It is quite possible this is an
issue with the server you are trying to connect to and not a client
issue.

Note that the ssh2 maintainer is extremely busy and is the only person
maintaining ssh2, so he can take a bit of time to respond. The bit
challenge for him is that if he cannot reproduce the issue, he cannot
debug it and therefore cannot fix it. As I mentioned, I have used ssh2
and ssh2-sftp-client to connect to servers with over 1000 files in a
directory with no problems. If you can provide him with sufficient
information to identify the problem, he usually provides an update
fairly quickly.

Finally, a new version of ssh2 was just released. I have pushed out
ssh2-sftp-client 7.0.2, which is based on this new ssh2 version. You
will need to upgrade and verify the issue persists. (I expect it will,
but it needs to be demonstrated anyway).

**Comment by bmmarko** [comment]
Hello everyone! I'm happy that I came across this thread just 2 days after starting to use this package.

I experienced the same issue today. `Error: list: Packet length 45555 exceeds max length of 34000`

After some testing with different conditions I came to conclusion that this issue might be coming from negotiation between client & server.

I have two SFTP servers, with different server software. On one of them, listing directory with over 1K files works perfectly well, while other one throws the error when attempting to list far less files (less than 500).

Looking forward for any solution.

**Comment by theophilusx** [comment]
I think your probably correct.

This is an issue in ssh2 and nothing can be done from ssh2-sftp-client
level.

Could you please add this information to issue msceex/ssh2#1048?

Please ensure you include node version, client platform and any details
you have about the failing sftp server. The more people we can show
affected by this will raise the importance of the issue and get
attention earlier.

**Comment by calebjacobson** [comment]
ssh2 has pushed version 1.4.0 which addresses this issue [change](https://github.com/mscdex/ssh2/commit/4606d0e127e3d27ecde1850a2649f33eef9970ab), [version bump](https://github.com/mscdex/ssh2/commit/cde0b02b8ee7d641559c4bfcb317b3839884897a). Can you update to using ssh2 1.4.0?

**Comment by theophilusx** [comment]
Yes, I was just waiting for 1.4.0 to be released. Will push a new
version tomorrow.
